### PR TITLE
feat: verification for Polkadot address

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [x] Feature

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Related to https://github.com/kodadot/private-workers/issues/180
- [x] Requires deployment https://github.com/kodadot/private-workers/pull/182
- [x] Closes https://github.com/kodadot/nft-gallery/issues/10471

[ Currently, only support Polkadot address ]

Add verified process before the below action: create profile / edit profile / follow / unfollow

It would ask for signing a message on the wallet extension before action. The verified message would be saved on localstorage forever unless you disconnect wallet / switch account.

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by prury** [comment]
nice, thank you!

File: components/profile/create/Modal.vue [hunk 1]
```diff
@@ -42,6 +42,7 @@ import { rateLimitedPinFileToIPFS } from '@/services/nftStorage'
 import { appClient, createChannel } from '@/services/farcaster'
 import { StatusAPIResponse } from '@farcaster/auth-client'
 import { useDocumentVisibility } from '@vueuse/core'
+import { SIGNATURE_MESSAGE } from '@/utils/constants'
```
Comment (hassnian) [inline components/profile/create/Modal.vue:45]: no need, can be auto-imported

File: components/profile/create/Modal.vue [hunk 2]
```diff
@@ -121,7 +127,9 @@ const processProfile = async (profileData: ProfileFormData) => {
 const handleFormSubmition = async (profileData: ProfileFormData) => {
   stage.value = 4 // Go to loading stage
   try {
-    await processProfile(profileData)
+    const signature = await getSignedMessage()
```
Comment (hassnian) [inline components/profile/create/Modal.vue:130]: can we move this inside `processProfile`

File: services/profile.ts [hunk 3]
```diff
@@ -61,6 +65,15 @@ export type UpdateProfileRequest = {
 export type FollowRequest = {
   initiatorAddress: string
   targetAddress: string
+  signature: string
+  message: string
+}
+
+const invalidSignatureErrorHandler = (error: FetchError) => {
+  if (error.status === 401) {
+    useWalletStore().setSignedMessage('')
```
Comment (hassnian) [inline services/profile.ts:74]:
```suggestion
setSignedMessage(
// null 
// undefined
) 
```

**Review by hassnian — COMMENTED** [review]
personal preference but I'm not a fan of the constant import of `SIGNATURE_MESSAGE` in multiple places , I'd use it just inside the `useVerifyAccount` where we can retrieve a pair with the message and the signature, if we need to change something it's just in one place

quick POC, feel free to change names  

CleanShot 2024-06-19 at 11 33 15@2x

**Comment by vikiival** [comment]
cc @Jarsen136

**Review by hassnian — APPROVED** [review]

**Comment by Jarsen136** [comment]
> @Jarsen136 missing image / video

video added

**Comment by Jarsen136** [comment]
Updated: disable profile / follow on base network

**Comment by vikiival** [comment]
This is imo a misleading

Screenshot 2024-06-19 at 18 46 39

**Pull request merged**